src/Swarm/Game/Robot.hs

```diff
@@ -191,16 +215,19 @@ type Robot = RobotR Identity
 --   directly reference fields inside this record; for example, one
 --   can use 'robotName' instead of writing @'robotEntity'
 --   . 'entityName'@.
-robotEntity :: Lens' Robot Entity
+robotEntity :: Lens' (RobotR phase) Entity
 
 -- | The creation date of the robot.
 robotCreatedAt :: Lens' Robot TimeSpec
 
--- | The name of a robot.  Note that unlike entities, robot names are
---   expected to be globally unique
+-- | The name of a robot.
 robotName :: Lens' Robot Text
 robotName = robotEntity . entityName
 
+-- | The name of a robot template.
+trobotName :: Lens' TRobot Text
+trobotName = robotEntity . entityName
```

Comment:
Why do we need a separate lens for the name? It should not depend on phase. :thinking:

Reply:
Maybe you do not want to expose lenses with the `RobotR phase` type?

If the idea is to hide it, then why export it?

Reply:
We don't really need it of course, but if we make it polymorphic then type inference starts failing at use sites with horrible error messages like
```
 Could not deduce (Control.Effect.Sum.Member
                          (State (RobotR phase5)) sig)
        arising from a use of ‘.=’
```
Of course we could fix this by putting type annotations or explicit type applications whenever we use `robotName`, but I thought this seemed nicer.  I will add a comment explaining this.